guides/source/getting_started.md

```diff
@@ -7,1150 +7,1551 @@ This guide covers getting up and running with Ruby on Rails.
+```erb#6,8
+<h1>Products</h1>
+
+<div id="products">
+  <% @products.each do |product| %>
+    <div>
+      <a href="/products/<%= product.id %>">
+        <%= product.name %>
+      </a>
+    </div>
+  <% end %>
+</div>
+```
+
+Refresh this page in your browser and you'll see that this works, but we can do better.
+
+Rails provides helper methods for generating paths and URLs. When you run `rails routes`, you'll see the Prefix column. This prefix matches the helpers you can use for generating URLs with Ruby code.
+
+```
+                                  Prefix Verb   URI Pattern                                                                                       Controller#Action
+                                products GET    /products(.:format)                                                                               products#index
+                                 product GET    /products/:id(.:format)                                                                           products#show
+```
+
+These route prefixes give us helpers like the following:
+
+* `products_path` generates `"/products`"`
+* `products_url` generates `"http://localhost:3000/products`"`
+* `product_path(1)` generates `"/products/1"`
+* `product_url(1)` generates `"http://localhost:3000/products/1"`
+
+`_path` returns a relative path which the browser understands is for the current domain.
+`_url` returns a full URL including the protocol, host, and port.
+
+URL helpers are useful for rendering emails that will be viewed outside of the browser.
+
+Combined with the `link_to` helper, we can generate anchor tags and use the URL helper to do this cleanly in Ruby. `link_to` accepts the display content for the link (`product.name`)and the path or URL to link to for the `href` attribute (`product`).
+
+Let's refactor this to use these helpers:
+
+```erb#6
+<h1>Products</h1>
+
+<div id="products">
+  <% @products.each do |product| %>
+    <div>
+      <%= link_to product.name, product %>
+    </div>
+  <% end %>
+</div>
+```
+
+### Creating Products
+
+So far we've had to create products in the Rails console, but let's make this work in the browser.
+
+We need to create two actions for create:
+
+1. The new product form to collect product information
+2. The create action in the controller to save the product and check for errors
+
+Let's start with our controller actions.
+
+```ruby
+class ProductsController < ApplicationController
+  def index
+    @products = Product.all
+  end
+
+  def show
+    @product = Product.find(params[:id])
+  end
+
+  def new
+    @product = Product.new
+  end
+end
+```
+
+The `new` action instantiates a new `Product` which we will use for displaying the form fields.
+
+We can update `app/views/products/index.html.erb` to link to the new action.
+
+```erb#3
+<h1>Products</h1>
+
+<%= link_to "New product", new_product_path %>
+
+<div id="products">
+  <% @products.each do |product| %>
+    <div>
+        <%= link_to product.name, product %>
```

Comment:
Looks like 4 space indentation here instead of just 2 :)

Reply:
Good catch! We had a mix of tabs and spaces from earlier reviews, so this probably left over from converting those.